versioned_hdf5/subchunk_map.py

```diff
@@ -184,9 +382,81 @@ def chunk_submap(self, chunk_idx: hsize_t) -> tuple[Slice, slice, slice]:
             slice(chunk_sub_start, chunk_sub_stop, sel_step),
         )
 
+    @cython.cfunc
+    @cython.nogil
+    @cython.exceptval(check=False)
+    def _read_many_slices_param(
+        self,
+        chunk_idx: hsize_t,
+    ) -> tuple[hsize_t, hsize_t, hsize_t, hsize_t]:
+        chunk_start, chunk_stop = self._chunk_start_stop(chunk_idx)
+        sel_start = self.start
+        sel_stop = self.stop
+        sel_step = self.step
+
+        abs_start = max(chunk_start, sel_start)
+        # Get the smallest lcm multiple of common that is >= start
+        abs_start = smallest_step_after(abs_start, sel_start % sel_step, sel_step)
+
+        # shift start so that it is relative to index
+        value_sub_start = (abs_start - sel_start) // sel_step
+        value_sub_stop = ceil_a_over_b(min(chunk_stop, sel_stop) - sel_start, sel_step)
+
+        count = value_sub_stop - value_sub_start
+        chunk_sub_start = abs_start - chunk_start
+        chunk_sub_stride = sel_step
```

Comment:
copy paste of `chunk_submap()` immediately above. No point to refactor to deduplicate code as `chunk_submap()` will be deleted.